drivers/ipm/ipm_handlers.c

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <syscall_handler.h>
+#include <ipm.h>
+
+_SYSCALL_HANDLER(ipm_send, dev, wait, id, data, size)
+{
+	_SYSCALL_OBJ(dev, K_OBJ_DRIVER_IPM);
+	_SYSCALL_MEMORY_READ(data, size);
+	return _impl_ipm_send((struct device *)dev, wait, id,
+			      (const void *)data, size);
+}
```

Comment:
This is another where I think it might be best to leave it out.  The quark se IPM mechanism is very low level and there's nothing here to prevent e.g. a user process with access to this API from terribly confusing whatever protocol is being sent over that pipe.

Basically: an app that cares about protecting the system from the behavior of a process wouldn't give that process this API to begin with.

Reply:
I think this might be OK to leave in?
A user thread won't be able to use these APIs unless granted permissions on the device objects that represents the IPM channel. So a random thread won't be able to muck about with the protocol, only those threads that have been blessed with access.
I'm thinking of uses where user threads could be on the sending end of the pipe, shouldn't need to be supervisor to do that.